src/tests/UnitTests/EnsureEnumParamTests.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using EnsureThat;
+using Xunit;
+
+namespace UnitTests
+{
+    public class EnsureEnumParamTests : UnitTestBase
+    {
+        [Fact]
+        public void IsDefined_ShouldNotThrow()
+        {
+            var item = Only1IsValid.Valid;
+
+            ShouldNotThrow(
+                () => Ensure.Enum.IsValidEnum(item, ParamName),
+                () => EnsureArg.IsValidEnum(item, ParamName),
+                () => Ensure.That(item, ParamName).IsValidEnum());
+        }
+
+        [Fact]
+        public void IsNotDefined_ShouldThrow()
+        {
+            var item = (Only1IsValid)2;
+
+            ShouldThrow<ArgumentOutOfRangeException>(
```

Comment:
I do understand that they will differ, but that's also part of the purpose of having different API's. But what do you think about:

`Ensure.Enum.Contains`

`EnsureArg.EnumContains`

`Ensure.That(x).IsDefined()`

Reply:
I can do that if you wish, but it seems to differ from the precedent for other APIs -- e.g. `HasItems`. Is there another API that similarly has all different? If not I'd encourage avoiding 'surprising' people.

If you want the above still on further consideration just say something like "I think it's not too surprising" :smile:

Reply:
`Type` has `IsEnumDefined` that the IsDefined ends up calling, what about that name for all (even though it's redundant for Ensure.Enum)

Reply:
- [ ] Contains, EnumContains, IsDefined
- [ ] all IsEnumDefined
- [ ] all IsDefined

Reply:
The API's don't have to be aligned. If naming etc works in all contexts, that would of course be preferred. How-ever, I generally don't like to repeat the context in the name.

Thinking of it the first `Ensure.Enum` and third context `Ensure.That` can be aligned. It's the middle one that benefits from having context in the name. So I do believe the following would be rather intuitive, no?

`Ensure.Enum.IsDefined(..)`
`EnsureArg.EnumIsDefined(..)`
`Ensure.That(..).IsDefined()`